docs/advanced/server-side-rendering.md

```diff
@@ -177,254 +177,156 @@ we'll have to import from. `preact-ssr-prepass`.
+The next step is to query data in your client components by means of the `useQuery`
+method defined in `@urql/next`.
 
-```js
-import { withUrqlClient, initUrqlClient } from 'next-urql';
-import { ssrExchange, dedupExchange, cacheExchange, fetchExchange, useQuery } from 'urql';
-import { executeExchange } from '@urql/exchange-execute';
-
-import { schema } from '@/server/graphql'; // our GraphQL server's executable schema
+```tsx
+// app/client/page.tsx
+'use client';
 
-const TODOS_QUERY = `
-  query { todos { id text } }
-`;
+import Link from 'next/link';
+import { Suspense } from 'react';
+import { useQuery, gql } from '@urql/next';
 
-function Todos() {
-  const [res] = useQuery({ query: TODOS_QUERY });
+export default function Page() {
   return (
-    <div>
-      {res.data.todos.map(todo => (
-        <div key={todo.id}>
-          {todo.id} - {todo.text}
-        </div>
-      ))}
-    </div>
+    <Suspense>
+      <Pokemons />
+    </Suspense>
   );
 }
 
-export async function getServerSideProps(ctx) {
-  const ssrCache = ssrExchange({ isClient: false });
-  const client = initUrqlClient(
-    {
-      url: '', // not needed without `fetchExchange`
-      exchanges: [
-        dedupExchange,
-        cacheExchange,
-        ssrCache,
-        executeExchange({ schema }), // replaces `fetchExchange`
-      ],
-    },
-    false
-  );
-
-  await client.query(TODOS_QUERY).toPromise();
+const PokemonsQuery = gql`
+  query {
+    pokemons(limit: 10) {
+      id
+      name
+    }
+  }
+`;
 
-  return {
-    props: {
-      urqlState: ssrCache.extractData(),
-    },
-  };
+function Pokemons() {
+  const [result] = useQuery({ query: PokemonsQuery });
+  return (
+    <main>
+      <h1>This is rendered as part of SSR</h1>
+      <ul>
+        {result.data.pokemons.map((x: any) => (
+          <li key={x.id}>{x.name}</li>
+        ))}
+      </ul>
+    </main>
+  );
 }
-
-export default withUrqlClient(ssr => ({
-  url: 'your-url',
-}))(Todos);
 ```
 
-Direct schema execution skips one network round trip by accessing your resolvers directly
-instead of performing a `fetch` API call.
+The data queried in the above component will be rendered on the server
+and re-hydrated back on the client. When using multiple Suspense boundaries
+these will also get flushed as they complete and re-hydrated.
 
-### Stale While Revalidate
+> When data is used throughout the application we advise against
+> rendering this as part of a server-component so you can benefit
+> from the client-side cache.
 
-If we choose to use Next's static site generation (SSG or ISG) we may be embedding data in our initial payload that's stale on the client. In this case, we may want to update this data immediately after rehydration.
-We can pass `staleWhileRevalidate: true` to `withUrqlClient`'s second option argument to Switch it to a mode where it'll refresh its rehydrated data immediately by issuing another network request.
+### Invalidating data from a server-component
```

Comment:
fwiw, this doc goes back and forth between "server-component" and "server component" in case you wanted to consolidate to one or the other